prow/plugins/lifecycle/lifecycle.go

```diff
@@ -0,0 +1,113 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package lifecycle
+
+import (
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/plugins"
+)
+
+const deprecatedWarn = true
+
+var (
+	deprecatedTick = time.Tick(time.Hour) // Warn once per hour
+	lifecycleRe    = regexp.MustCompile(`(?mi)^/(remove-)?(freeze|stale|putrid|rotten|frozen)( cancel)?\s*$`)
```

Comment:
Why do we want both `/remove-X` as well as `/X cancel`?

Reply:
What if I `/remove-freeze cancel`?

Reply:
Nevermind, saw the handler logic -- are we supporting an older workflow with `/remove-X`? Seems like `/X cancel` has been our common pattern lately.

Reply:
There is also `/remove-kind`, `/remove-sig`, and `/remove-area`.

Reply:
:(

Reply:
We should probably standardize on one or the other pattern everywhere.

Reply:
add a TODO or comment here?